pkg/cmd/extension/browse/browse.go

```diff
@@ -0,0 +1,527 @@
+		Qualifiers: search.Qualifiers{
+			Topic: []string{"gh-extension"},
+		},
+	})
+	if err != nil {
+		return extEntries, fmt.Errorf("failed to search for extensions: %w", err)
+	}
+
+	host, _ := opts.Cfg.DefaultHost()
+
+	for _, repo := range result.Items {
+		if !strings.HasPrefix(repo.Name, "gh-") {
+			continue
+		}
+		ee := extEntry{
+			URL:         "https://" + host + "/" + repo.FullName,
+			FullName:    repo.FullName,
+			Name:        repo.Name,
+			description: repo.Description,
+		}
+		for _, v := range installed {
+			// TODO consider a Repo() on Extension interface
+			var installedRepo string
+			if u, err := git.ParseURL(v.URL()); err == nil {
+				if r, err := ghrepo.FromURL(u); err == nil {
+					installedRepo = ghrepo.FullName(r)
+				}
+			}
+			if repo.FullName == installedRepo {
+				ee.Installed = true
+			}
+		}
+		if repo.Owner.Login == "cli" || repo.Owner.Login == "github" {
+			ee.Official = true
+		}
+
+		extEntries = append(extEntries, ee)
+	}
+
+	return extEntries, nil
+}
+
+func ExtBrowse(opts ExtBrowseOpts) error {
+	if opts.Debug {
+		f, err := os.CreateTemp("", "extBrowse-*.txt")
+		if err != nil {
+			return err
+		}
+		defer os.Remove(f.Name())
+
+		opts.Logger = log.New(f, "", log.Lshortfile)
+	} else {
+		opts.Logger = log.New(io.Discard, "", 0)
+	}
+
+	opts.IO.StartProgressIndicator()
+	extEntries, err := getExtensions(opts)
+	opts.IO.StopProgressIndicator()
+	if err != nil {
+		return err
+	}
+
+	opts.Rg = newReadmeGetter(opts.Client, time.Hour*24)
+
+	app := tview.NewApplication()
+
+	outerFlex := tview.NewFlex()
+	innerFlex := tview.NewFlex()
+
+	header := tview.NewTextView().SetText(fmt.Sprintf("browsing %d gh extensions", len(extEntries)))
+	header.SetTextAlign(tview.AlignCenter).SetTextColor(tcell.ColorWhite)
+
+	filter := tview.NewInputField().SetLabel("filter: ")
+	filter.SetFieldBackgroundColor(tcell.ColorGray)
+	filter.SetBorderPadding(0, 0, 20, 20)
+
+	list := tview.NewList()
+
+	readme := tview.NewTextView()
+	readme.SetBorderPadding(1, 1, 0, 1)
+	readme.SetBorder(true).SetBorderColor(tcell.ColorPurple)
+
+	help := tview.NewTextView()
+	help.SetText(
+		"/: filter  i/r: install/remove  w: open in browser  pgup/pgdn: scroll readme  q: quit")
+	help.SetTextAlign(tview.AlignCenter)
+
+	ui := uiRegistry{
+		App:       app,
+		Outerflex: outerFlex,
+		List:      list,
+	}
+
+	extList := newExtList(opts, ui, extEntries)
+
+	loadSelectedReadme := func() {
+		rendered, err := getSelectedReadme(opts, readme, extList)
+		if err != nil {
+			opts.Logger.Println(err.Error())
+			readme.SetText("unable to fetch readme :(")
+			return
+		}
+
+		app.QueueUpdateDraw(func() {
+			readme.SetText("")
+			readme.SetDynamicColors(true)
+
+			w := tview.ANSIWriter(readme)
+			_, _ = w.Write([]byte(rendered))
+
+			readme.ScrollToBeginning()
+		})
+	}
+
+	filter.SetChangedFunc(func(text string) {
+		extList.Filter(text)
+		go loadSelectedReadme()
+	})
+
+	filter.SetDoneFunc(func(key tcell.Key) {
+		switch key {
```

Comment:
What do you think about allowing the down key to also do `extList.Focus()`? I naturally reached for that key after I was done typing in my search filter.